mautic/mautic
# Global search fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [Y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/pull/11435



#### Description:

https://github.com/mautic/mautic/pull/9693 introduced a bug that broke the global search. There is another fix for that in https://github.com/mautic/mautic/pull/11435 which is fixing the symptom, not the cause. I suggest this fix as adding the join could have break other areas where the `getEntities()` method is called like API and who knows what else. I suggest to add the join only when needed.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test steps from https://github.com/mautic/mautic/pull/9693
3. Test steps from https://github.com/mautic/mautic/pull/11435
